hilti/runtime/include/types/stream.h

```diff
@@ -1280,16 +1268,23 @@ class View {
     View limit(const Offset& incr) const { return View(begin(), begin() + incr); }
 
     /**
-     * Extracts a fixed number of stream from the view.
+     * Extracts a fixed number of stream bytes from the view.
      *
-     * @tparam N number of stream to extract
      * @param dst target array to write stream data into
+     * @param n number of stream bytes to extract
      * @return new view that has it's starting position advanced by N
      */
-    template<int N>
-    View extract(Byte (&dst)[N]) const {
+    View extract(Byte* dst, uint64_t n) const {
         _ensureValid();
-        return View(SafeConstIterator(detail::extract<N>(dst, unsafeBegin(), unsafeEnd())), _end);
+
+        if ( n > size() )
+            throw WouldBlock("end of stream view");
+
+        auto p = unsafeBegin();
+        for ( int i = 0; i < n; ++i )
+            dst[i] = *p++;
+
+        return View(SafeConstIterator(p), _end);
     }
```

Comment:
While we are at it, let's move this into the `.*cc` file.

Reply:
I left this in the header to the compiler a chance to inline.